src/Native/Runtime/thread.cpp

```diff
@@ -1284,9 +1284,14 @@ COOP_PINVOKE_HELPER(UInt8*, RhCurrentNativeThreadId, ())
 }
 #endif // CORERT
 
+extern "C" int InitializeRuntime();
+
 // Standard calling convention variant and actual implementation for RhpReversePInvokeAttachOrTrapThread
 EXTERN_C NOINLINE void FASTCALL RhpReversePInvokeAttachOrTrapThread2(ReversePInvokeFrame * pFrame)
 {
+    // Entrypoint for reverse PInvoke
+    // ensure runtime is initialized
+    InitializeRuntime();
```

Comment:
I would push it even lower into `Thread::ReversePInvokeAttachOrTrapThread` to the path that is initializing the thread.